code/modules/mob/mob_helpers.dm

```diff
@@ -248,6 +248,12 @@
 	firstname.Find(real_name)
 	return firstname.match
 
+/// Find the last name of a mob from the real name with regex
+/mob/proc/last_name()
+	var/static/regex/lasttname = new("\[^\\s-\]+$") //First word before whitespace or "-"
+	lasttname.Find(real_name)
+	return lasttname.match
```

Comment:
Are you sure this didn't already exist?

I remember that heirlooms already were able to pull out the last name of "Lou Briccant"

Reply:
the heirloom component just does something along the lines of `var/list/names = splittext(name, " ")` followed by **var/surname = names[names.len]**. I feel like this one is more consistent with `first_name` and also safer to use on hyphenated names.